src/main/java/org/tugraz/sysds/hops/codegen/SpoofCompiler.java

```diff
@@ -602,6 +606,13 @@ private static void rConstructModifiedHopDag(Hop hop,  HashMap<Long, Pair<Hop[],
 			//replace sub-dag with generated operator
 			Pair<Hop[], Class<?>> tmpCla = clas.get(hop.getHopID());
 			CNodeTpl tmpCNode = cplans.get(hop.getHopID()).getValue();
+
+			if (DMLScript.LINEAGE) {
+				//construct and save lineage DAG from pre-modification HOP DAG
+				LineageItem LIroot = LineageItemUtils.ConstructLineageFromHops(hop);
```

Comment:
Yes, you can integrate it here, but you might want to consider caching, as the same generated operator might be used multiple times in the program. So maybe before constructing it, try to probe the cache. 

Also, please start method names with a small character (constructLineageFromHops)